### ISSUE TITLE
Include Pocket Video Feed

### ISSUE THREAD

**Issue by bbinto** [mozilla-mobile/firefox-tv#659]
User Stories:
- As a user, I want to open Firefox to be entertained by interesting recommended web video content
- As a user, I want to have one place for discovering and viewing web video content on my TV

More [here](https://docs.google.com/document/d/1icOKaOhkTymWen6K37tqKGpH_J56X6Hk2GrSedBp_ok/edit#heading=h.hdve4bbf99u3)

**Comment by liuche** [comment]
Adding @lime124  and @BrianNJones to take a look at the list - _nothing actionable_ but these are the things I can think of for what Eng would need for Pocket feeds so giving a heads up.

**Comment by liuche** [comment]
Starting a comment for task breakdown, starting with a list of things from an eng perspective:

*UX*
- [ ] (low-fi) mocks for positioning of Pocket Videos on home screen (for ref here's the [Android TV Recommendations](https://designguidelines.withgoogle.com/android-tv/system-overview/home.html) row spec) [how many videos, above/below other bundled tiles]
- [ ] interactions available to Pocket tiles (remove? show as not pinned?)
- [ ] Onboarding? Any explanation of Pocket feed? (or maybe "recommendations" are expected on tv apps)
- [ ] Loading/placeholder animations? if API fetch is slow

*Copy*
- [ ] Title for Pocket recommendations section
- [ ] Empty state (if API call not working)

*Eng*
- [ ] Add Pocket API secrets for release-build-only (Fennec uses TC Secrets and preprocessing) - maybe manual for now?
- [ ] TBD implementing mocks

**Comment by mcomella** [comment]
@bbinto Is there an existing web version of the Pocket video feed? I wonder if we could use that instead of building new native UI. Some benefits of web:
- We can change it without shipping a new release
- There are fewer expectations around offline access and performance (e.g. no caching code, which is often the most tricky to get right)
- Supposedly modifying prototypes in web code is faster than modifying mobile code... (if you're experienced)

Downsides:
- The web site would use the cursor (or we'd have to implement spatial navigation)
- The UI wouldn't be like other apps so we'd probably have to make it seem like we navigated to the website in the browser, which might be less attractive to users
- We're not web devs so we might need other people to help us, assuming we'd even have access to the source. And we'd have to learn new tools (e.g. web telemetry)

**Comment by bbinto** [comment]
@st3fan's team is maybe more advanced in their Pocket video feed integration for Activity Stream?

I like where you are going with this, especially re: "Supposedly modifying prototypes in web code is faster than modifying mobile code... (if you're experienced)"

Downsides are def not negligible either, especially the need to use the cursor.

**Comment by mkoidin** [comment]
I'm not aware of anywhere that we are publishing on the web the global videos feed currently. We have used it in our mobile apps, but that was built natively. I <think> one of the other Mozilla teams integrated it (Android maybe?) - but I'm not sure what approach they took.

I'm not super familiar with the specifics of what you are dealing with on the Fire TV side of things, but I will say that generally when we've built something with a web view, we've later reverted to going native for performance and UI purposes.

**Comment by Sdaswani** [comment]
I think I’m ambivalent about web vs native, but I would want to do something *compelling*, e.g., if we surface the top three pocket vids on the home screen, can we have them play on the home screen (maybe without sound) when the user travels over it. 

But I agree with @mkoidin that type of experience is _usually_ easier to implement and/or better if you go native.

**Comment by liuche** [comment]
I don't think this is a "web vs native" question, but rather a "is there something we can reuse" - it seems like there isn't any existing thing (other than what @mcomella already built) so let's just wait for UX to come up with mocks.

**Comment by liuche** [comment]
Btw, if this helps UX, including an excerpt of what the Pocket Video API returns.

Basically they include:
* url
* excerpt (description of the video)
* domain (like Youtube - actually they all look like Youtube)
* image (link to thumbnail to include - we probably should also have some kind of "default" if this link is broken/empty)
* miscellaneous Pocket-specific stuff (id, dedupe url hosted on pocket, timestamp)

```
{"id":13548,
"url":"https:\\/\\/pocket.co\\/xQDbu8",
"dedupe_url":"https:\\/\\/www.youtube.com\\/watch?v=EL1wN_ihL_E",
"title":"Jurassic '
 'Park Auditions - SNL",
"excerpt":"Bill Hader hosts Saturday Night Live on '
 'March 17, 2018, with musical guest Arcade Fire.#SNL #SNL43Get more SNL: '
 'http:\\/\\/www.nbc.com\\/saturday-night-liveFull Episodes: '
 'http:\\/\\/www.nbc.com\\/saturday-night-liv...Like SNL: '
 'https:\\/\\/www.facebook.com\\/snlFollow SNL: '
 'https:\\/\\/twitter.com\\/nbcsnlSNL Tumblr: '
 'http:\\/\\/",
"domain":"youtube.com",
"image_src":"https:\\/\\/img-getpocket.cdn.mozilla.net\\/direct?url=http%3A%2F%2Fimg.youtube.com%2Fvi%2FEL1wN_ihL_E%2Fmaxresdefault.jpg&resize=w450",
"published_timestamp":"0","sort_id":0},

{"id":13543,
"url":"https:\\/\\/pocket.co\\/xQDbuI",
"dedupe_url":"https:\\/\\/www.youtube.com\\/watch?v=trWLY6NrS2Q",
"title":"Honest '
 'Trailers - Every Wes Anderson Movie",
"excerpt":"Before you see the latest '
 "animated feature from your barista's favorite director, relive his "
 'meticulous works from the past that made you kind of happy, kind of sad, and '
 "kind of unsure - It's Every Wes Anderson Movie!Watch the Honest Trailer "
 'Commentaries to hear the writers thoughts on the movies '
 'al",
"domain":"youtube.com",
"image_src":"https:\\/\\/img-getpocket.cdn.mozilla.net\\/direct?url=http%3A%2F%2Fimg.youtube.com%2Fvi%2FtrWLY6NrS2Q%2Fmaxresdefault.jpg&resize=w450","published_timestamp":"0","sort_id":1},
```

**Comment by bbinto** [comment]
Some notes from the UX meeting

For v1:
- need to design Pocket Video feed list (native vs. web)
- no personalized feeds yet (no login)

To discuss
- Frequency of pulling from Pocket API
- Most visited, or most viewed

**Comment by mkoidin** [comment]
@bbinto Apologies for the delayed response - I was on PTO the past two weeks.

Re: the JSON response - if there is something else needed (or that we want to remove), we can tweak that - so happy to discuss.
